Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -2041,6 +2048,11 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
 
         this._layoutBadgeElement.addEventListener("click", this._layoutBadgeClicked.bind(this), true);
         this._layoutBadgeElement.addEventListener("dblclick", this._layoutBadgeDoubleClicked, true);
+
+        if (!hadLayoutBadge) {
+            this.representedObject.addEventListener(WI.DOMNode.Event.LayoutOverlayShown, this._updateLayoutBadgeStatus, this);
+            this.representedObject.addEventListener(WI.DOMNode.Event.LayoutOverlayHidden, this._updateLayoutBadgeStatus, this);
+        }
```

Comment:
Can this be an `else` off of :2022-2028 instead? It would be clearer IMO if the adding/removing of the layout event listeners were together such that one or the other happens. This function overall is very hard to grok, and bringing those calls closer together will make it easier to do so.

Reply:
i cant make it an `else` as that would go against the styleguide (since the `if` has a `return`), but i can move this up so that they're closer together :)